rasa/nlu/test.py

```diff
@@ -1128,7 +1129,7 @@ def generate_folds(
 
     skf = StratifiedKFold(n_splits=n, shuffle=True)
     x = td.intent_examples
-    y = [example.get("intent") for example in x]
+    y = [example.get_combined_intent_response_key() for example in x]
```

Comment:
```suggestion
    ## Get labels with response key appended to intent name because we want a stratified split on all intents(including retrieval intents if they exist)
    y = [example.get_combined_intent_response_key() for example in x]
```